x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/ShowColumns.java

```diff
@@ -82,23 +84,23 @@ public void execute(SqlSession session, ActionListener<Page> listener) {
             List<List<?>> rows = emptyList();
             if (indexResult.isValid()) {
                 rows = new ArrayList<>();
-                fillInRows(indexResult.get().mapping(), null, rows);
+                fillInRows(indexResult.get().mapping(), null, session.configuration().version(), rows);
             }
             listener.onResponse(of(session, rows));
         }, listener::onFailure));
     }
 
-    private void fillInRows(Map<String, EsField> mapping, String prefix, List<List<?>> rows) {
+    static void fillInRows(Map<String, EsField> mapping, String prefix, SqlVersion version, List<List<?>> rows) {
         for (Entry<String, EsField> e : mapping.entrySet()) {
             EsField field = e.getValue();
             DataType dt = field.getDataType();
             String name = e.getKey();
-            if (dt != null) {
+            if (dt != null && isTypeSupportedInVersion(dt, version)) {
```

Comment:
Same question here: if we show unsupported fields, we should do the same for supported types when reporting to old clients.

Reply:
Removed the filtering in `SHOW COLUMNS`. Note that this is still kept in `SYS COLUMNS`, to prevent (some) clients from automatically selecting these columns.